graphql/src/main/java/com/scalar/db/graphql/schema/TableGraphQlModel.java

```diff
@@ -0,0 +1,447 @@
+        .type(getPayloadObjectType)
+        .argument(newArgument().name("get").type(nonNull(getInputObjectType)))
+        .build();
+  }
+
+  private GraphQLEnumType createClusteringKeyNameEnum() {
+    GraphQLEnumType.Builder builder = newEnum().name(objectType.getName() + "_ClusteringKeyName");
+    tableMetadata.getClusteringKeyNames().forEach(builder::value);
+    return builder.build();
+  }
+
+  private GraphQLInputObjectType createClusteringKeyInputObjectType() {
+    return newInputObject()
+        .name(objectType.getName() + "_ClusteringKey")
+        .field(newInputObjectField().name("name").type(nonNull(clusteringKeyNameEnum)))
+        .field(newInputObjectField().name("intValue").type(Scalars.GraphQLInt))
+        .field(newInputObjectField().name("floatValue").type(Scalars.GraphQLFloat))
+        .field(newInputObjectField().name("stringValue").type(Scalars.GraphQLString))
+        .field(newInputObjectField().name("booleanValue").type(Scalars.GraphQLBoolean))
+        .build();
+  }
+
+  private GraphQLInputObjectType createOrderingInputObjectType() {
+    return newInputObject()
+        .name(objectType.getName() + "_Ordering")
+        .field(newInputObjectField().name("name").type(nonNull(clusteringKeyNameEnum)))
+        .field(newInputObjectField().name("order").type(nonNull(typeRef("Order"))))
+        .build();
+  }
+
+  private GraphQLInputObjectType createScanInputObjectType() {
+    return newInputObject()
+        .name(objectType.getName() + "_ScanInput")
+        .field(
+            newInputObjectField().name("partitionKey").type(nonNull(partitionKeyInputObjectType)))
+        .field(
+            newInputObjectField().name("start").type(list(nonNull(clusteringKeyInputObjectType))))
+        .field(
+            newInputObjectField()
+                .name("startInclusive")
+                .type(Scalars.GraphQLBoolean)
+                .defaultValueLiteral(new BooleanValue(true)))
+        .field(newInputObjectField().name("end").type(list(nonNull(clusteringKeyInputObjectType))))
+        .field(
+            newInputObjectField()
+                .name("endInclusive")
+                .type(Scalars.GraphQLBoolean)
+                .defaultValueLiteral(new BooleanValue(true)))
+        .field(newInputObjectField().name("orderings").type(list(orderingInputObjectType)))
+        .field(newInputObjectField().name("limit").type(Scalars.GraphQLInt))
+        .field(newInputObjectField().name("consistency").type(typeRef("Consistency")))
+        .build();
+  }
+
+  private GraphQLObjectType createScanPayloadObjectType() {
+    return newObject()
+        .name(objectType.getName() + "_ScanPayload")
+        .field(
+            newFieldDefinition()
+                .name(objectType.getName())
+                .type(nonNull(list(nonNull(objectType)))))
+        .build();
+  }
+
+  private GraphQLFieldDefinition createQueryScanField() {
+    return newFieldDefinition()
+        .name(objectType.getName() + "_scan")
```

Comment:
Can you clarify the naming rules?
For this,  it is not capitalized (`_scan`) but for others, it's like `_ScanPayload`.
I see `_put` and `_delete` so there seem some rules behind but I'm not sure and it looks inconsistent.

Reply:
* All generated names for a table are prefixed with the table name. (The object type name (`objectType.getName()`) is the table name itself.)
* “Types” such as `_ScanPayload`, `_GetInput`, or `_Key` are in pascal case.
* “Fields” in the `Query` and `Mutation` GraphQL type (`_get`, `_scan`, `_put`, and `_delete`) are in lowercase.

Reply:
Thank you!
Using different case styles for `types` and `fields` is a common practice?
It would be great if we have good reasons for those rules (ideally described in the code) since it looks inconsistent at first glance.

Reply:
> Using different case styles for types and fields is a common practice?

For example, here is Apollo's recommendation. I think this convention is very common.
https://www.apollographql.com/docs/apollo-server/schema/schema/#naming-conventions

The difference in our app is that we simply prefix all types and field names with the `<table_name>` + `_`.